lib/scripting/post-process/expr.ts

```diff
@@ -17,138 +17,271 @@
+	let index = node.children[0].text.length;
+	for (let loop = 1; loop < node.children.length; loop++) {
+		if (node.text.substr(index).startsWith(' Is ')) {
+			estree = binaryExpression('==', estree, node.children[loop].estree);
+			index += node.children[loop].text.length + 4;
+		}
+	}
+	return estree;
 }
 
-export function mult(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const operator = result[2].text as BinaryOperator;
-	const rightExpr = result[4];
-	return estree.binaryExpression(operator, leftExpr, rightExpr);
+function ppUnaryExpression(node: ESIToken): any {
+	return unaryExpression(node.text.charAt(0) as UnaryOperator, node.children[0].estree);
 }
 
-export function unary(result: [Token, null, Expression]): Expression {
-	const operator = result[0].text as UnaryOperator;
-	const expr = result[2];
-	return estree.unaryExpression(operator, expr);
+function ppLogicalNotExpression(node: ESIToken): any {
+	return unaryExpression('!', node.children[0].estree);
 }
 
-export function concat(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const operator = result[2].text as BinaryOperator;
-	const rightExpr = result[4];
-	return estree.binaryExpression('+', leftExpr, rightExpr);
+function ppParenthesizedExpression(node: ESIToken): any {
+	return node.children[1].estree;
 }
 
-export function is(result: [Expression, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[4];
-	return estree.binaryExpression('==', leftExpr, rightExpr);
+function ppMemberAccessExpression(node: ESIToken) {
+	return identifier('.' + node.children[1].estree.name);
 }
 
-export function isNot(result: [Expression, null, Token, null, Token, null, Expression]): Expression {
-	const leftExpr = result[0];
-	const rightExpr = result[6];
-	return estree.binaryExpression('!=', leftExpr, rightExpr);
+function ppSubExpression(node: ESIToken): any {
+	let id: any = null;
+	const argLists = [];
+	let expr = null;
+	let args;
+	for (const child of node.children) {
+		if (child.type === 'MemberAccessExpression' || child.type === 'SimpleNameExpression') {
+			id = child.estree;
+		} else if (child.type === 'OpenParenthesis') {
+			args = [];
+		} else if (child.type === 'ArgumentList') {
+			args = child.estree;
+		} else if (child.type === 'CloseParenthesis') {
+			argLists.push(args);
+		} else if (child.type === 'SubExpression') {
+			expr = child.estree;
+		}
+	}
+	let estree: any;
+	if (argLists.length > 0) {
+		for (const argList of argLists) {
+			estree = callExpression(estree ? estree : id, argList);
+		}
+	} else {
+		estree = id;
+	}
+	if (expr != null) {
+		let found = false;
+		estree = replace(expr, {
+			enter: astNode => {
+				if (!found) {
+					if (astNode.type === 'Identifier') {
+						found = true;
+						return memberExpression(estree, identifier(astNode.name.substr(1)));
+					}
+				}
+			},
+		});
```

Comment:
A brief explanation what this does would be nice

Reply:
Oh boy.. 😃 This deals with dot identifiers and with statements.  `a.b.c.d.e.f()`, `a().b.c.d.e().f.g()` , `.a.b.c.d.e.f()`, When the subexpression gets an additional subexpression, we only need to modify the first found identifier (when walking the tree)  And since the it will usually be a member expression, we need to remove the period from the identifier name. 

I wasn't sure if you can break out of a replace. That was the purpose of the found flag. 

BTW, we also need to keep the very first dot, for use in with statements.